causal_testing/surrogate/surrogate_search_algorithms.py

```diff
@@ -0,0 +1,224 @@
+from causal_testing.specification.causal_specification import CausalSpecification
+from causal_testing.testing.estimators import Estimator, PolynomialRegressionEstimator
+from causal_testing.surrogate.causal_surrogate_assisted import SearchAlgorithm, SearchFitnessFunction
+
+from pygad import GA
+from operator import itemgetter
+import multiprocessing as mp
+import warnings
+
+contradiction_functions = {
+    "positive": lambda x: -1 * x,
+    "negative": lambda x: x,
+    "no_effect": lambda x: abs(x),
+    "some_effect": lambda x: abs(1 / x),
+}
+
+
+class GeneticSearchAlgorithm(SearchAlgorithm):
+    def __init__(self, delta=0.05, config: dict = None) -> None:
+        super().__init__()
+
+        self.delta = delta
+        self.config = config
+        self.contradiction_functions = {
+            "positive": lambda x: -1 * x,
+            "negative": lambda x: x,
+            "no_effect": lambda x: abs(x),
+            "some_effect": lambda x: abs(1 / x),
+        }
+
+    def generate_fitness_functions(
+        self, surrogate_models: list[PolynomialRegressionEstimator]
+    ) -> list[SearchFitnessFunction]:
+        fitness_functions = []
+
+        for surrogate in surrogate_models:
+            contradiction_function = self.contradiction_functions[surrogate.expected_relationship]
+
+            def fitness_function(_ga, solution, idx):
```

Comment:
ga and idx are not used in this function which is the final Pylint issue. Are they needed? If so it would be worth commenting why and we can add a pylint ignore line

Reply:
`# pylint: disable=unused-argument` used on this line and a comment is added above to justify their lack of use